srtcore/core.cpp

```diff
@@ -7569,6 +7569,13 @@ void srt::CUDT::bstats(CBytePerfMon *perf, bool clear, bool instantaneous)
         perf->pktRcvUndecryptTotal  = m_stats.rcvr.undecrypted.total.count();
         perf->byteRcvUndecryptTotal = m_stats.rcvr.undecrypted.total.bytes();
 
+        
+        // Average values management
+        m_stats.sndr.fulfillMeasuresTable(perf->pktSent, double(perf->byteSent));
+        m_stats.rcvr.fulfillMeasuresTable(perf->pktRecv, double(perf->byteRecv));
+        perf->mbpsSendRate        = m_stats.sndr.getAverageValueFromTable();
+        perf->mbpsRecvRate        = m_stats.rcvr.getAverageValueFromTable();
```

Comment:
Calculation of the average sending and receiving rates better not depend on the frequency the API function `srt_stats` called. It can be called every 1s, every 100 ms, just once in a while or even by multiple users in a cloud deployment or REST API & Graphana scenario.
I would suggest counting packets sent on every packet being sent.

Reply:
Won't it use to much CPU to do this on every packet ? The sample period is 10ms. Maybe we can have a thread that fulfill the table every 10ms and only keep these data in stats ?
        perf->mbpsSendRate        = m_stats.sndr.getAverageValueFromTable();
        perf->mbpsRecvRate        = m_stats.rcvr.getAverageValueFromTable();

Reply:
Well, it should be just dereferencing the array value and adding to it (O(n)). Higher burden than in your suggestions, but I don't think that high to consider. Can be tested though. If too much burden, the sampling can be done discretely, say, every 100 packets or so.
The main thing is that `bstats` is called from outside, and there is no restriction on the frequency or the number of calls. Will it work as good if called every 100 ms? BTW, having some test results would be great.

Reply:
I though on it too and I know how I can do easily to get what you want :). The method will work in all cases because it computes directly the difference between two calls to select the right slot. It's only the fill method that needs to be called somewhere else and the fill method that I need to rework to increment an already fill slot. But I have it :)